src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -795,9 +833,11 @@ private void retrieveTags(final BitbucketSCMSourceRequest request) throws IOExce
     @Override
     protected SCMRevision retrieve(SCMHead head, TaskListener listener) throws IOException, InterruptedException {
         final BitbucketApi bitbucket = buildBitbucketClient();
-        List<? extends BitbucketBranch> branches = bitbucket.getBranches();
         if (head instanceof PullRequestSCMHead) {
             PullRequestSCMHead h = (PullRequestSCMHead) head;
+            List<BitbucketBranch> branches = new ArrayList<>();
+            branches.addAll(bitbucket.getBranchesByFilterText(h.getBranchName()));
```

Comment:
here you are triggering two REST call to bitbucket cloud, it was better before, it takes all branches and filter here based on the logic you need

Reply:
the main problem was load on bitbucket server, yes, it added an additional request for small repos which did not contain more branches than configured in paging size, the original request just returns all branches

Reply:
we have monorepositories (and their fork) with each one 6/7 pages in branch list in bitbucket. It's preferred get all branches and filter than here. Since we are speaking to reduce calls to avoid reach user call limits, these changes act in constrast with the goal of JENKINS-65909

Reply:
but basically line 839 is just not needed in this case, as 840 does exactly the same as the loop in findCommit, it reduces the list of branches to the ones matching the name so in the end the list currently contains two branches (source and target)